new/multiple-policy-revision.md

```diff
@@ -0,0 +1,110 @@
+software, the MySQL install and configuration is manage by the DBA team while
+ntpd and rsyslog are managed by a core infrastructure team. These live in
+different cookbooks so for the most part things are copacetic. When the DBA
+team wants to roll out a new MySQL configuration entry they update the cookbook,
+recompile the policyfile, `chef push` that compiled policy to the QA policy
+group, and then follow the usual Policyfile roll-out procedure. For the core
+team, things are more complex. They could follow the same process, but this
+would require careful coordination with the DBA team to ensure they don't have
+a roll-out in progress (see https://yolover.poise.io/#21). They could also
+release their cookbook changes either to a git branch or Supermarket server,
+and then notify the DBA team that they should run a roll-out. Neither of these
+are wonderful options.
+
+## Motivation
+
+   As a maintainer of core, company-level cookbooks,
+    I want to release cookbook updates,
+    so that I can deploy changes quickly.
+
+With that said, a more specific case that isn't currently handled well:
+
+The Database team owns MySQL and uses Chef to deploy it along with related configs
+and tools. The Monitoring team owns Collectd, which is deployed to every server
+in the organization, and they are responsible for configuring collectd and deploying
+new versions of it as needed. The core team owns ntpd and is responsible for its
+configuration.
+
+All teams want to follow the snapshot-based approach where they have a mono-repo
+of their own cookbooks with a few shared cookbooks pulled in via dependencies.
+When a team wants to do a release they run `chef update` to recompile their
+policy, and then use `chef push` to roll it out to each environment in sequence
+(with time in-between for testing et al). The teams also do not want to have to
+ask permission from any other team when deploying new versions of a service they
+are responsible for.
+
+This combination of a snapshot-based workflow without explicit team coordination
+is not currently an easy thing to do with the policy systems.
+
+
+## Specification
+
+The proposed solution to the motivation described above is to allow a more composable form of policy which would be able to optionally include other policies inside it.
+
+This could be done with the addition of a ```include_policy``` directive, an example of which is shown below:
+
+```ruby
+include_policy "base", git: "github.com/myorg/policies.git", path: "foo/bar/baz.lock.json"
+```
+
+The ```include_policy``` directive will support three sources for policies: 
+
+* ```git```, with the following parameter being the URL to a Git repository and a path parameter specifying the location of the file within the repository
+* ```server```, with the following paramater being set to a Chef server URL, a policy_name paramter and a policy_revision_id parameter.
+* ```local```, with the following parameter being a path to a file on disk.
+
```

Comment:
http support for people building policy files to and storing them in artifactory or s3 (for example) would be good.

Reply:
I think that's an entirely reasonable thing - I'm probably not going to target it for the initial RFC, but it's absolutely something we could look at adding in.